daemon/daemon.go

```diff
@@ -299,6 +302,64 @@ func (daemon *Daemon) LogToDisk(src *broadcastwriter.BroadcastWriter, dst, strea
 	return nil
 }
 
+type LineWriter struct {
+	buf  *bytes.Buffer
+	next io.WriteCloser
+}
+
+func NewLineWriter(w io.WriteCloser) *LineWriter {
+	return &LineWriter{
+		buf:  bytes.NewBuffer(nil),
+		next: w,
+	}
+}
+
+func (w *LineWriter) Write(p []byte) (n int, err error) {
+	w.buf.Write(p)
+
+	for {
+		if newlineIdx := bytes.IndexByte(w.buf.Bytes(), '\n'); newlineIdx == -1 {
+			break
+		}
+
+		// TODO: implement limiting line length
+
+		line, _ := w.buf.ReadString('\n')
+		line = strings.Trim(line, "\r\n")
+		w.next.Write([]byte(line))
+	}
+
+	return len(p), nil
+}
+
+func (w *LineWriter) Close() error {
+	return w.next.Close()
+}
+
+func (daemon *Daemon) LogToSyslog(src *broadcastwriter.BroadcastWriter, syslogConfig runconfig.SyslogConfig) error {
+	var (
+		log io.WriteCloser
+		err error
+	)
+	if syslogConfig.Url != "" {
+		u, err := url.Parse(syslogConfig.Url)
```

Comment:
Doesn't this redeclare 'err' within local scope and shadow the function's err checked outside?